### PR TITLE
Logerrors

### PR DESCRIPTION
**Pull request by btimby**
This PR address #31 by introducing a setting. `ELASTICSEARCH_DSL_AUTOSYNC_LOG_ERRORS`. This setting is `False` by default (current functionality is preserved). However, when enabled, errors are logged (but not raised) by the `BaseSignalProcessor` and it's subclasses.

I want this option to allow myself to have best effort real time indexing without sinking the ship for every indexing error. This option in combination with periodic rebuild of the indexes should provide the trade-off I am looking for.

### CODE REVIEW

**Comment by safwanrahman** [comment]
This is interesting! can you explain the use case a bit more?

**Comment by btimby** [comment]
Like #31, my case is that I don't want an ElasticSearch problem to affect database transactions. I don't want an exception in the signal handler to report an error or fail to make the change.

IOW, I want ES errors to be ignored (but logged) in the signal handler. Then I have a cron job that rebuilds the index periodically (to resolve these errors).

This is best-effort real-time synchronization with periodic full synchronization.

File: django_elasticsearch_dsl/signals.py [hunk 1]
```diff
@@ -54,22 +66,42 @@ def handle_save(self, sender, instance, **kwargs):
...
     def handle_delete(self, sender, instance, **kwargs):
         """Handle delete.
 
         Given an individual model instance, delete the object from index.
         """
-        registry.delete(instance, raise_on_error=False)
+        try:
+            registry.delete(instance)
+
+        except Exception:
```
Comment (safwanrahman) [inline django_elasticsearch_dsl/signals.py]: Whats the reason behind capturing all the `Exception`? I think you can catch only the needed exceptions like above.

Comment (btimby) [reply django_elasticsearch_dsl/signals.py]: The reasoning is to preserve what the method did previously. It passed a flag: `raise_on_error=False`. I removed that (so errors are raised, so I can log them). Then I emulate what used to happen by stuffing `Exception`.

Comment (sabricot) [reply django_elasticsearch_dsl/signals.py]: I am agree with @safwanrahman, the `raise_on_error` is used in https://elasticsearch-py.readthedocs.io/en/master/helpers.html#elasticsearch.helpers.bulk and by looking to the code of this method i think we only need to catch  https://github.com/elastic/elasticsearch-py/blob/de8413fdbdc6a207401a8cdc8091bf1dda319dab/elasticsearch/helpers/__init__.py#L13

**Pull request closed without merging**